p2p/src/authenticated/actors/tracker/address_record.rs

```diff
@@ -0,0 +1,86 @@
+use crate::authenticated::types::SignedPeerInfo;
+use commonware_cryptography::Scheme;
+use std::net::SocketAddr;
+
+/// If the count for an `AddressRecord` is set to this value,
+/// it is considered pinned and cannot be incremented or decremented.
+const PINNED: usize = usize::MAX;
+
+/// Represents information known about a peer's address.
+#[derive(Clone)]
+pub enum AddressRecord<C: Scheme> {
+    /// Provided during initialization.
+    /// Can be upgraded to `Discovered`.
+    Bootstrapper(SocketAddr),
+
+    /// Peer address is not yet known.
+    /// Can be upgraded to `Discovered`.
+    /// Tracks the number of peer sets this peer is part of.
+    Unknown(usize),
+
+    /// Discovered this peer's address from other peers.
+    /// Tracks the number of peer sets this peer is part of.
+    Discovered(usize, SignedPeerInfo<C>),
+}
+
+impl<C: Scheme> AddressRecord<C> {
+    /// Get the address of the peer.
+    ///
+    /// Returns None if the address is unknown.
+    pub fn get_address(&self) -> Option<SocketAddr> {
+        match &self {
+            Self::Bootstrapper(socket) => Some(*socket),
+            Self::Discovered(_, peer_info) => Some(peer_info.socket),
+            Self::Unknown(_) => None,
+        }
+    }
+
+    /// Attempt to set the address of a discovered peer.
+    ///
+    /// Returns true if the update was successful.
+    /// Panics if the address is a bootstrapper.
```

Comment:
We may receive signed info over the wire for a bootstrapper (they may not be synchronized across network participants). We shouldn't panic in this case.

Reply:
Ah you're right. Old comment once I realized my mistake I changed the logic but not the comment. Removed